src/tests/Microsoft.PowerFx.Interpreter.Tests/ConfigTests.cs

```diff
@@ -861,6 +862,52 @@ public void MutableSupportedFunctionsTest(string functionName, string expression
             Assert.Contains(checkFalse.Errors, e => e.MessageKey == "ErrUnknownFunction" && e.Message.Contains($"'{functionName}' is an unknown or unsupported function."));
         }
 
+        [Theory]
+        [InlineData(300, "Text(DateTimeValue(\"2023-12-21T12:34:56.789Z\"), DateTimeFormat.UTC)", true)]
+        [InlineData(10, "Text(DateTimeValue(\"2023-12-21T12:34:56.789Z\"), DateTimeFormat.UTC)", false)]
+        [InlineData(200, "Len(With({one: \"aaaaaaaaaaaaaaaaaa\"}, Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(one, \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3)))", false)]
+        [InlineData(500, "Len(With({one: \"aaaaaaaaaaaaaaaaaa\"}, Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(Substitute(one, \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3), \"a\", one, 3)))", true)]
+        public void AllowedMaxExpressionLengthTest(int maxLength, string expression, bool isSucceeded)
+        {
+            var config = new PowerFxConfig
+            {
+                MaximumExpressionLength = maxLength
+            };
+
+            var engine = new Engine(config);            
+            var opt = engine.GetDefaultParserOptionsCopy();
+            var check = engine.Check(expression);
+
+            if (isSucceeded)
+            {
+                Assert.True(check.IsSuccess);
+                Assert.True(expression.Length < opt.MaxExpressionLength);
+            }
+            else
+            {
+                Assert.False(check.IsSuccess);
+                Assert.False(expression.Length < opt.MaxExpressionLength);
+            }
+        }
+
+        [Fact]
+        public void NotAllowedMaxExpressionLengthTest()
+        {
+            var config = new PowerFxConfig();
+            int strLength = config.MaximumExpressionLength + 20;
+            StringBuilder expression = new StringBuilder();
+
+            for (uint i = 0; i < strLength; i++)
+            {
+                expression.Append("a");
+            }
+
+            var engine = new Engine(config);
+            var check = engine.Check(expression.ToString());
+
+            Assert.False(check.IsSuccess);
```

Comment:
this will always fail, regardless of length, since 'aaaaaa' is an invalid identifier. 
If we wrap as a string (quotes on both side), then it's valid.

Reply:
removed this one. thanks